packages/@n8n/nodes-langchain/nodes/vendors/GoogleGemini/helpers/utils.ts

```diff
@@ -165,3 +225,100 @@ export async function transferFile(
 
 	return { fileUri: file.uri, mimeType: file.mimeType };
 }
+
+export async function createFileSearchStore(
+	this: IExecuteFunctions,
+	displayName: string,
+): Promise<IDataObject> {
+	return (await apiRequest.call(this, 'POST', '/v1beta/fileSearchStores', {
+		body: { displayName },
+	})) as IDataObject;
+}
+
+export async function uploadToFileSearchStore(
+	this: IExecuteFunctions,
+	i: number,
+	fileSearchStoreName: string,
+	displayName: string,
+	downloadUrl?: string,
+	fallbackMimeType?: string,
+	qs?: IDataObject,
+) {
+	const fileData = await getFileStreamFromUrlOrBinary.call(
+		this,
+		i,
+		downloadUrl,
+		fallbackMimeType,
+		qs,
+	);
+
+	let stream: Stream;
+	let mimeType: string;
+
+	if ('buffer' in fileData) {
+		stream = Readable.from(fileData.buffer);
+		mimeType = fileData.mimeType;
+	} else {
+		stream = fileData.stream;
+		mimeType = fileData.mimeType;
+	}
+
+	const uploadResponse = (await uploadStream.call(this, stream, {
+		endpoint: `/upload/v1beta/${fileSearchStoreName}:uploadToFileSearchStore`,
+		mimeType,
+		body: { displayName, mimeType },
+	})) as { body: { name: string } };
+
+	const operationName = uploadResponse.body.name;
+	let operation = (await apiRequest.call(
```

Comment:
There are a lot of type casts, here it would be better to define the type like

```ts
let operation: FileSearchOperation = await apiRequest.call(...);
```

This also removes the need to cast in https://github.com/n8n-io/n8n/pull/22988/files#diff-600f79f07af7d7d6a14c5c378f09e1c1b39b11900b8b833f48208fb687263347R285